src/Middleware/ResponseCompression/test/ResponseCompressionMiddlewareTest.cs

```diff
@@ -592,7 +592,7 @@ public async Task FlushHeaders_SendsHeaders_Compresses(string encoding, int expe
                     {
                         context.Response.Headers[HeaderNames.ContentMD5] = "MD5";
                         context.Response.ContentType = TextPlain;
-                        context.Response.Body.Flush();
+                        await context.Response.Body.FlushAsync();
```

Comment:
The next test already covers FlushAsync. This test should opt into sync IO so it can test Flush.

Reply:
Thanks. Should have paid more attention here.
Do you think I should do the same for the other tests. Eg. copy them and create both synchronous and async versions?

Reply:
I'm not as concerned about them, their code paths are much simpler. Compression is several layers deep so it's more important that we test both code paths.